index.js

```diff
@@ -78,5 +83,182 @@ function loader (mcVersion) {
+      if (itemOne.name === 'enchanted_book' && itemTwo.displayName !== 'Enchanted Book') throw new Error('Can only combine book with book')
+      else if (itemOne.name === 'enchanted_book') return // return here because this func will throw because enchanted book isnt fixable
+      let [, fixMaterials] = canFixData.find(([items]) => items.includes(itemOne.displayName))
+      fixMaterials = fixMaterials.concat(['Enchanted Book', itemOne.displayName])
+      // if (itemOne.name === 'enchanted_book' && itemTwo.displayName !== 'Enchanted Book') throw new Error('Can only combine book with book')
+      if (!fixMaterials.includes(itemTwo.displayName)) throw new Error('Not able to be combined')
+    }
+    /**
+     *
+     * @param {Item} itemOne left hand item
+     * @param {Item} itemTwo right hand item
+     * @returns {[xpLevelCost, fixedDurability, usedMats]}
+     * xpLevelCost is the number of xp levels used for repair (if any)
+     * fixedDurability is duribility after using the anvil
+     * usedMats is the number of materials used to fix the broken item (if many mats is used)
+     */
+    function repairCost (itemOne, itemTwo) {
+      const MAX_DURABILITY = maxItemDura[itemOne.name]
+      const { metadata: DURABILITY_LOST, displayName: toolToBeFixed } = itemOne
+      const { displayName: fixMaterial } = itemTwo
+      const [, fixMaterials] = canFixData.find(([items]) => items.includes(toolToBeFixed)) // can always be fixed by another of itself
+      if (!fixMaterials.includes(fixMaterial) && toolToBeFixed !== fixMaterial) {
+        return 0 // Enchanted book can't fix
+      }
+      if (itemOne.name === 'enchanted_book') return 0 // not fixable
+      // result vars
+      let fixedDurability = 0
+      let xpLevelCost = 0
+      let usedMats = 0
+      if (itemTwo.name === itemOne.name) {
+        fixedDurability = (0.12 * MAX_DURABILITY) + itemTwo.metadata
+        xpLevelCost = 2
+        usedMats = 1
+      } else if (DURABILITY_LOST !== 0) {
+        const durabilityFixedPerMat = Math.floor(MAX_DURABILITY * 0.25)
+        const matsToFullyRepair = Math.ceil(DURABILITY_LOST / durabilityFixedPerMat)
+        if (itemTwo.count > matsToFullyRepair) { // takeall of itemTwo
+          fixedDurability = MAX_DURABILITY
+          xpLevelCost = matsToFullyRepair // 1 exp lvl per mat used
+          usedMats = matsToFullyRepair
+        } else {
+          fixedDurability = DURABILITY_LOST + (durabilityFixedPerMat * itemTwo.count)
+          xpLevelCost = itemTwo.count // 1 exp lvl per mat used
+          usedMats = itemTwo.count
+        }
+      }
+      return [xpLevelCost, fixedDurability, usedMats]
+    }
+
+    /**
+     *
+     * @param {Item} itemOne
+     * @param {Item} itemTwo
+     * @param {boolean} creative
+     * @param {boolean} rename
+     */
+    function combine (itemOne, itemTwo, creative, rename) {
+      const data = {}
+      try {
+        combinePossible(itemOne, itemTwo) // throws if not possible
```

Comment:
document this if it throws

Reply:
what do you mean by document?

Reply:
Add it in the docs that this can function throw

Reply:
It doesn't actually throw, the way enchants work is if it throws it returns 0, because that means on the client to show the red X instead of allow anvil'ing